File: frontend/types/pairing.ts
export type CreatePairCodeResponse = {
  ok: boolean;
  code?: string;
  expires_at?: string;
  connected?: boolean;
  error?: string;
};

export type PairStatusResponse = {
  ok: boolean;
  code?: string;
  connected?: boolean;
  device_name?: string | null;
  expires_at?: string;
  used?: boolean;
  error?: string;
};

export type DebuggerConnectResponse = {
  ok: boolean;
  message?: string;
  device_name?: string;
  code?: string;
  error?: string;
};

File: styles/schemas/pairing.py
from pydantic import BaseModel
from typing import Optional


class CreatePairCodeResponse(BaseModel):
    ok: bool
    code: str
    connected: bool
    device_name: Optional[str]
    expires_at: str
    used: bool


class PairStatusResponse(BaseModel):
    ok: bool
    code: str
    connected: bool
    device_name: Optional[str]
    expires_at: str
    used: bool
    error: Optional[str] = None


class AgentRegisterRequest(BaseModel):
    code: str
    device_name: str
